api/envoy/config/route/v3/route_components.proto

```diff
@@ -837,6 +837,62 @@ message RouteAction {
   string prefix_rewrite = 5
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
 
+  // Indicates that some headers received from upstream cluster need to be rewritten. When upstream
+  // returns a header, the header will be overwritten with the given value before being sent back as
+  // response to the client.
+  //
+  // In the below example there are two response header rewrites mentioned: one for location header
+  // and one for x-random-token header.
+  //
+  // The location header's value will be matched with the given match pattern as a prefix to the header's
```

Comment:
You probably want to use `envoy.type.v3.StringMatcher` for this, rather than hardcode prefix match semantics.